### PR TITLE
feat: Add table, matrix, sheet, and view icons

### PR DESCRIPTION
**Pull request by zenoamaro**
## Tables & Views

This is a set of related icons to represent tables, grids, matrices for rich-text editing, data manipulation, database administration, development tools.

### Table

* Icon name: `table`
* Use case: tables in rich text editors; database tables
* Screenshots of similar icons: (see below)

### View

* Icon name: `view` or `observable` or `observe` or `spy` or `eye-box`
* Use case: database views and real-time computed objects; sensors; view and focus modes; observing modes
* Screenshots of similar icons: (see below)

### Matrix

* Icon name: `matrix`
* Use case: calculation grids and matrices (but not layouts)
* Screenshots of similar icons: (see below)

### Sheet

* Icon name: `sheet` or `data` or `dataset` or `data-sheet`
* Use case: labeled collections of data
* Screenshots of similar icons: (see below)

## Proposal

Icons are based on `columns` (included for reference and visual continuity). SVG and Figma file are attached below. Icon `view` was revised by @jletey.

Original size

Embiggened

[Table, matrix, sheet, view.zip](https://github.com/featherity/featherity/files/4782632/Table.matrix.sheet.view.zip)

## Encounters in the wild

Screenshot 2020-06-13 at 02 46 45

Screenshot 2020-06-13 at 02 46 58

### CODE REVIEW

**Comment by ericfennis** [comment]
Nice work!

Is the eye from the view icon the same that `eye.svg`?

**Comment by zenoamaro** [comment]
@ericfennis, yes, it is a shrunk-down version of `eye.svg`.

**Comment by locness3** [comment]
Maybe `grid`would suit better than `matrix`?

**Comment by locness3** [comment]
Also, I'm not sure what the view icon would be used for compared to the èye`one

**Comment by jletey** [comment]
> Maybe `grid` would suit better than `matrix`?

I agree, but there is already a [`grid`](https://github.com/featherity/featherity/blob/master/icons/grid.svg) icon. Although I'm not rlly a fan of this one 😕

**Comment by locness3** [comment]
I would rename the current one `apps` or smth like this tbh

**Comment by zenoamaro** [comment]
> Maybe `grid` would suit better than `matrix`?

I chose `matrix` because: 

1. `grid` is already in use
2. it is unambiguously a data/science term

The current `grid` is usually called `layout-grid` or `view-grid` or similar, but it's probably still a more common interpretation than "apps".

> Also, I'm not sure what the view icon would be used for compared to the eye one

The main semantic difference is that `eye` is a state, or an action; while `view` is an object or entity, usually (as in my case) a "virtual slice" of another database table or entity.

I listed use-cases as: 

* database views
* real-time computed objects 
* view modes, focus modes
* sensors

To be fair, `view` is quite a valuable term, so it's ok to rename this to `observable` or even `eye-box`. But, as it is, I think it already works well for any meaning of the term `view`.

**Comment by locness3** [comment]
I'd love eye-box, and I'd rename matrix to grid (and grid to layout-grid) to accomodate more use cases

**Comment by zenoamaro** [comment]
I would be completely fine with `eye-box`. Actually, that's how I might start searching for it.

About `grid`: here are various searches across all of react-icon's icon-sets. I think consensus is split about whether it should be like feather's grid, or this matrix:

Screenshot 2020-06-17 at 18 20 51
Screenshot 2020-06-17 at 18 21 01
Screenshot 2020-06-17 at 18 21 08
Screenshot 2020-06-17 at 18 21 20

File: icons/sheet.svg [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  xmlns="http://www.w3.org/2000/svg"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M3 9h18M3 9v6m0-6V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2v4m0 0v6M3 15v4a2 2 0 0 0 2 2h4m-6-6h18m0 0v4a2 2 0 0 1-2 2h-4m-6 0h6m-6 0V9m6 12V9" />
```
Comment (ericfennis) [inline icons/sheet.svg:12]: Here as well
```suggestion
  <rect x="3" y="3" width="18" height="18" rx="2" ry="2"/>
  <line x1="3" y1="9" x2="21" y2="9" />
  <line x1="3" y1="15" x2="21" y2="15" />
  <line x1="9" y1="9" x2="9" y2="21" />
  <line x1="15" y1="9" x2="15" y2="21" />
```

File: icons/matrix.svg [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  xmlns="http://www.w3.org/2000/svg"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M3 9h18M3 9v6m0-6V5a2 2 0 0 1 2-2h4m12 6v6m0-6V5a2 2 0 0 0-2-2h-4M3 15v4a2 2 0 0 0 2 2h4m-6-6h18m0 0v4a2 2 0 0 1-2 2h-4M9 3v18M9 3h6M9 21h6m0-18v18" />
```
Comment (ericfennis) [inline icons/matrix.svg:12]: For example this one in much easier to read than a path element.
```suggestion
  <rect x="3" y="3" width="18" height="18" rx="2" ry="2"/>
  <line x1="3" y1="9" x2="21" y2="9" />
  <line x1="3" y1="15" x2="21" y2="15" />
  <line x1="9" y1="3" x2="9" y2="21" />
  <line x1="15" y1="3" x2="15" y2="21" />
```

File: icons/view.svg [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M5 12s2.545-5 7-5c4.454 0 7 5 7 5s-2.546 5-7 5c-4.455 0-7-5-7-5z" />
+  <path d="M12 13a1 1 0 1 0 0-2 1 1 0 0 0 0 2z" />
+  <path d="M21 8V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v3m18 8v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-3" />
```
Comment (ericfennis) [inline icons/view.svg:14]: Those are tricky ones.
But maybe the circle in the eye.

Here some documentation to achieve that

**Comment by ericfennis** [comment]
@zenoamaro Any progress on this?

**Comment by zenoamaro** [comment]
> Maybe refactor the svgs a bit using [basic svg elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Basic_Shapes) as much as possible.
> Makes it more readable like the other feather icons.

I imagine most editing is done using illustration software (like I also used) which would destroy the carefully hand-crafted svg code. Is this reasonable to do?

**Comment by ericfennis** [comment]
> I imagine most editing is done using illustration software (like I also used) which would destroy the carefully hand-crafted svg code. Is this reasonable to do?

Yeah it can reduce the code size.
Makes the code also more readable.
Some illustration/design programs are not always export the SVG as small reduced code.

**Pull request merged**